bach/bach/dataframe.py

```
@@ -2015,6 +2016,62 @@ def mode(self, axis=1, skipna=True, level=None, numeric_only=False, **kwargs):
         return self._aggregate_func('mode', axis, level, numeric_only,
                                     skipna=skipna, **kwargs)
 
+    def quantile(
+        self,
+        q: Union[float, List[float]] = 0.5,
+        axis=1,
+        numeric_only=True,
+        **kwargs,
+    ):
+        """
+        Returns the quantile of all numeric columns.
+
+        :param q: value or list of values between 0 and 1.
+        :param axis: only ``axis=1`` is supported. This means columns are aggregated.
+        :param numeric_only: whether to aggregate numeric series only, or attempt all.
+        :returns: a new DataFrame with the aggregation applied to all selected columns.
+        """
+        df = self
+        if df.group_by is None:
+            df = df.groupby()
+
+        quantiles = [q] if isinstance(q, float) else q
+
+        all_quantile_dfs = []
+        for qt in quantiles:
+            new_series = df._apply_func_to_series(
+                func='quantile',
+                axis=axis,
+                numeric_only=numeric_only,
+                exclude_non_applied=True,
+                partition=df.group_by,
+                q=qt,
+                **kwargs,
+            )
+            initial_series = new_series[0]
```

question: do we not need to handle the two special cases where `len(q) == 0` and `len(self.data) == 0` ?

Or are we satisfied with raising an exception like we do now?

---

Let's add the checks indeed.

---

@jansentom we can raise an exception for now, Pandas just returns an empty  df for this cases, it just doesn't makes sense to me to try to calculate quantiles for a df that has no numeric series.
Also, I'll remove the `numeric_only` parameter